src/components/UserActions/UserActivity.ts

```diff
@@ -97,9 +108,13 @@ export class UserActivity extends Component {
 
         this.userProfileModel.getActions(this.options.userId).then((actions) => {
             const sortMostRecentFirst = (a: UserAction, b: UserAction) => b.timestamp.getTime() - a.timestamp.getTime();
-
             const sortedActions = actions.sort(sortMostRecentFirst);
```

Comment:
Since `.sort()` sorts the elements in-place, the `actions` argument is being altered in the process. I'd suggest to spread to a new array before sorting the entries.

```suggestion
            const sortedActions = [...actions].sort(sortMostRecentFirst);
```